File: Konfigurator-nestjs-Backend/konfigurator-nestjs-backend/src/dto/get-Highscores.dto.ts
import { ApiProperty} from '@nestjs/swagger';

export class GetHighscoresDto {

    constructor(
        date: string,
        gameMapId: number,
        username: string,
        score: number
        ) {
            this.date = date;
            this.gameMapId = gameMapId;
            this.username = username;
            this.score = score;
        }

        @ApiProperty()
        date: string;
    
        @ApiProperty()
        gameMapId: number;
    
        @ApiProperty()
        username: string;
    
        @ApiProperty()
        score: number;
}

File: Konfigurator-aspnetcore-Backend/Konfigurator-aspnetcore-Backend/Models/DTOs/GetHighscoresDto.cs
namespace Konfigurator_aspnetcore_Backend.Models.DTOs
{
    public class GetHighscoresDto
    {
        public GetHighscoresDto(DateTime date, int gameMapId, string username, int score)
        {
            this.date = date;
            this.gameMapId = gameMapId;
            this.username = username;
            this.score = score;
        }
        public DateTime date { get; }
        public int gameMapId { get; }
        public string username { get; }
        public int score { get; }
    }
}
